### PR TITLE
chore: allow type only imports for 'mongodb' package in connection-form

### PR DESCRIPTION
**Pull request by himanshusinghs**
## Description
This is a follow up for the [PR-5197](https://github.com/mongodb-js/compass/pull/5197) where we moved 'mongodb' to devDependencies in connection-form. In this PR we are restricting imports from 'mongodb' package to type only to avoid any possible regression in future.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by himanshusinghs** [comment]
By inline I meant this:
```
import { type X } from 'package-y';
```

We were earlier using the following syntax:
```
import type { X } from 'package-y';
```

but lately I saw a lot of usage of the inline type imports which sounds ok (maybe better in the way that they at-least allow import declaration of the package only once) but unfortunately that syntax fails for any of the rules (no-restricted-syntax, no-extraneous-dependencies, no-restricted-imports) that I tried while doing attempting this restriction.

**Comment by himanshusinghs** [comment]
Yea that sounds about right, the plugin would eventually get updated. Alright I will keep it as is. Thank you :)

File: packages/connection-form/.eslintrc.js [hunk 1]
```diff
@@ -5,4 +5,35 @@ module.exports = {
...
+        // Otherwise it would have been necessary to provide several polyfills for
+        // the dependencies of 'mongodb' package when webpacking it for an
+        // environment other than node. Hence we are restricting the usage of this
+        // package to type only imports.
+        '@typescript-eslint/no-restricted-imports': [
+          'error',
+          {
+            paths: [
+              {
+                name: 'mongodb',
```
Comment (gribnoysup) [inline packages/connection-form/.eslintrc.js:23]: I think connection storage was another one that we only want as type-only dep here, right?

**Review by gribnoysup — APPROVED** [review]

File: packages/connection-form/.eslintrc.js (hunk 1 above)
Comment (himanshusinghs) [reply packages/connection-form/.eslintrc.js:26]: Thank you, I overlooked this one - added this in [247e5ca](https://github.com/mongodb-js/compass/pull/5207/commits/247e5caf9a92f862aa0f26a65609785444867b14)

**Review by Anemy — APPROVED** [review]

File: packages/connection-form/.eslintrc.js [hunk 2]
```diff
@@ -5,4 +5,48 @@ module.exports = {
...
+                message: "Only type imports allowed from 'mongodb'",
+              },
+            ],
+            // Additionally we would like to make sure that connection-storage
+            // is not directly used in this package since it is very compass
+            // specific
+            patterns: [
+              {
+                group: ['@mongodb-js/connection-storage'],
+                message: 'Only type imports allowed from this package',
```
[the remark this answers is not in the document]
Comment (himanshusinghs) [reply packages/connection-form/.eslintrc.js:37]: Totally makes sense, and yea thanks for pointing out that I could combine both the packages in the same group 👍

**Pull request merged**